dkg/src/network_interface.rs

```diff
@@ -1,17 +1,52 @@
 // Copyright © Aptos Foundation
 
-use aptos_network::ProtocolId;
+use crate::DKGMessage;
+use aptos_config::network_id::{NetworkId, PeerNetworkId};
+use aptos_network::{
+    application::{error::Error, interface::NetworkClientInterface},
+    ProtocolId,
+};
+use aptos_types::PeerId;
+use std::time::Duration;
+
+pub const RPC: &[ProtocolId] = &[
+    ProtocolId::DKGRpcCompressed,
+    ProtocolId::DKGRpcBcs,
+    ProtocolId::DKGRpcJson,
+];
 
-/// Supported protocols in preferred order (from highest priority to lowest).
 pub const DIRECT_SEND: &[ProtocolId] = &[
     ProtocolId::DKGDirectSendCompressed,
     ProtocolId::DKGDirectSendBcs,
     ProtocolId::DKGDirectSendJson,
 ];
 
-/// Supported protocols in preferred order (from highest priority to lowest).
-pub const RPC: &[ProtocolId] = &[
-    ProtocolId::DKGRpcCompressed,
-    ProtocolId::DKGRpcBcs,
-    ProtocolId::DKGRpcJson,
-];
+#[derive(Clone)]
+pub struct DKGNetworkClient<NetworkClient> {
+    network_client: NetworkClient,
+}
+
+impl<NetworkClient: NetworkClientInterface<DKGMessage>> DKGNetworkClient<NetworkClient> {
+    /// Returns a new DKG network client
+    pub fn new(network_client: NetworkClient) -> Self {
+        Self { network_client }
+    }
+
+    /// Send a RPC to the destination peer
+    pub async fn send_rpc(
+        &self,
+        peer: PeerId,
+        message: DKGMessage,
+        rpc_timeout: Duration,
+    ) -> Result<DKGMessage, Error> {
+        let peer_network_id = self.get_peer_network_id_for_peer(peer);
+        self.network_client
+            .send_to_peer_rpc(message, rpc_timeout, peer_network_id)
+            .await
+    }
+
+    // TODO: we shouldn't need to expose this. Migrate the code to handle peer and network ids.
```

Comment:
What do you mean?

Reply:
it's copied from [here](https://github.com/aptos-labs/aptos-core/blob/63e5b9543c8ba4ad772d53a70f476159f4f07533/consensus/src/network_interface.rs#L173-L175)